### PR TITLE
testdisk: deduplicating testdisk/photorec

### PR DESCRIPTION
**Pull request by d-goldin**
###### Motivation for this change

Testdisk/Photorec has been packaged twice. This deduplicates
the packages by consolidating the packages into one and throwing
an error upon use of the outdated package.

this also adds qphotorec, which was previously not built and ensures
it's wrapped correctly.

Please note that I took the liberty to merge the maintainers lists.

See https://github.com/NixOS/nixpkgs/issues/67953

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @fgaz @edolstra

### CODE REVIEW

File: pkgs/tools/system/testdisk/default.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    pkgconfig
+  ]
+  ++ stdenv.lib.optional enableJpeg libjpeg
+  ++ stdenv.lib.optional enableZlib zlib
+  ++ stdenv.lib.optional enableNtfs ntfs3g
+  ++ stdenv.lib.optional enableExtFs e2fsprogs
+  ++ stdenv.lib.optional enableEwf libewf
+  ++ stdenv.lib.optional enableQt [ qtbase qttools qwt ];
+
+  nativeBuildInputs = stdenv.lib.optional enableQt [ wrapQtAppsHook ];
```
Comment (fgaz) [inline pkgs/tools/system/testdisk/default.nix]: Isn't  wrapQtAppsHook only necessary when not using the qt-specific mkDerivation?

Comment (d-goldin) [reply pkgs/tools/system/testdisk/default.nix]: Yep, turns out it's redundant.

File: pkgs/tools/system/testdisk/default.nix [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+assert enableJpeg -> libjpeg != null;
+assert enableZlib -> zlib != null;
+assert enableNtfs -> ntfs3g != null;
+assert enableExtFs -> e2fsprogs != null;
+assert enableEwf -> libewf != null;
+assert enableQt -> qtbase != null;
+assert enableQt -> qttools != null;
+assert enableQt -> qwt != null;
+
+mkDerivation rec {
```
Comment (fgaz) [inline pkgs/tools/system/testdisk/default.nix]: This brings in a qt dependency even if `enableQt = false`. It shoud be `(if enableQt then mkDerivation else stdenv.mkDerivation)`, I think

**Review by fgaz — CHANGES_REQUESTED** [review]

File: pkgs/tools/system/testdisk/default.nix [hunk 3]
```diff
@@ -0,0 +1,71 @@
+{ mkDerivation
+, stdenv
+, fetchurl
+, ncurses
+, libuuid
+, pkgconfig
+, enableJpeg ? false, libjpeg ? null
```
Comment (Mic92) [inline pkgs/tools/system/testdisk/default.nix]: Do we need so many options? Cannot users who want to disable `libjpeg` just do `testdisk.override { libjpeg = null; }`?

Comment (d-goldin) [reply pkgs/tools/system/testdisk/default.nix]: Yeah, they could. For some reason I was on some explicit train at that point. I will flip at least libjpeg.

**Review by fgaz — APPROVED** [review]
LGTM
thatnks for the cleanup, @d-goldin

**Comment by d-goldin** [comment]
@fgaz: Alright, thanks for the review and the good pointers. I was not very acquainted with the qt mechanisms. Let me know if I should squash or if merger could do it.

**Comment by fgaz** [comment]
@d-goldin Well, one way or another this will need to be squashed (or the commit messages edited to match the template), so you may as well do it

Also, while I'm at it:
@GrahamcOfBorg build testdisk testdisk-qt

**Review by fgaz — APPROVED** [review]

**Comment by d-goldin** [comment]
Is there anything left to do before this can be merged?

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -6409,7 +6409,7 @@ in
 
   termplay = callPackage ../tools/misc/termplay { };
 
-  testdisk-photorec = callPackage ../tools/system/testdisk-photorec { };
+  testdisk-photorec = throw "This package was a duplicate, please use testdisk or testdisk-qt instead";
```
Comment (vcunat) [inline pkgs/top-level/all-packages.nix]: Nitpick: perhaps use the usual `pkgs/top-level/aliases.nix`?

Comment (d-goldin) [reply pkgs/top-level/all-packages.nix]: @vcunat: Thanks for the pointer, I was actually not aware of this additional mechanism. Done.

**Comment by d-goldin** [comment]
Thanks.

File: pkgs/tools/system/testdisk/default.nix [hunk 5]
```diff
@@ -7,8 +7,8 @@
 , libjpeg
 , zlib
 , libewf
-, enableNtfs ? false, ntfs3g ? null
-, enableExtFs ? false, e2fsprogs ? null
+, enableNtfs ? stdenv.isDarwin, ntfs3g ? null
```
Comment (Mic92) [inline pkgs/tools/system/testdisk/default.nix]: Oh. This should be the opposite actually.

Comment (Mic92) [reply pkgs/tools/system/testdisk/default.nix]: fixed in b731394033435002ea3f93df7bfe9d05ed6e674f

**Pull request merged**